.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,203 @@
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Build artifacts
+        run: ./.github/workflows/scripts/build_artifacts.sh
+      - name: Archive artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: binaries
+          path: ./artifacts/
+
+  images:
+    runs-on: ubuntu-18.04
+    needs: [cache-deps]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Load cached build tools
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Build images
+        run: make images scratch-images
+      - name: Export images
+        run: docker save spire-server:latest-local spire-agent:latest-local k8s-workload-registrar:latest-local oidc-discovery-provider:latest-local spire-server-scratch:latest-local spire-agent-scratch:latest-local k8s-workload-registrar-scratch:latest-local oidc-discovery-provider-scratch:latest-local | gzip > images.tar.gz
+      - name: Archive images
+        uses: actions/upload-artifact@v2
+        with:
+          name: images
+          path: images.tar.gz
+
+  integration:
+    runs-on: ubuntu-18.04
+    needs: [cache-deps, images]
+    strategy:
+      fail-fast: false
+      matrix:
+        num_runners: [5]
+        runner_id: [1, 2, 3, 4, 5]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Load cached build tools
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Download archived images
+        uses: actions/download-artifact@v2
+        with:
+          name: images
+          path: .
+      - name: Load archived images
+        run: zcat images.tar.gz | docker load
+      - name: Run integration tests
+        env:
+          NUM_RUNNERS: ${{ matrix.num_runners }}
+          THIS_RUNNER: ${{ matrix.runner_id }}
+          TERM: dumb
+          CICD_TARGET_BRANCH: ${{ github.event.pull_request.base.ref }}
+        run: ./.github/workflows/scripts/split.sh | xargs ./test/integration/test.sh
+
+  publish-images:
+    runs-on: ubuntu-18.04
+    needs: [integration]
```

Comment:
I feel like most folks expect nightly to closely track main. Should we depend only on image build here? I think it's ok for nightly to break...? Relatedly, we've had integration breakages on several occasions that can be lengthy if it's due to upstream issues

Reply:
Sure, that's fair. I'll rip out lint/unit-test/integration-test and just keep artifact and image build.

Reply:
Actually I don't see a point in keeping artifact either.

Reply:
In fact, i just ripped out everything from nightly except a single job that builds and publishes the images under the `nightly` trag.